internal/agents/pi/model_routing_test.go

```
@@ -127,3 +122,301 @@ func TestResolvePackageBinErrors(t *testing.T) {
+	}
+}
+func TestModelRoutingClient_Contract(t *testing.T) {
+	cases := []struct{ op string; call func(c *ModelRoutingClient, ctx context.Context) error }{
+		{"capabilities", func(c *ModelRoutingClient, ctx context.Context) error { _, err := c.Capabilities(ctx); return err }},
+		{"inspect", func(c *ModelRoutingClient, ctx context.Context) error { _, err := c.Inspect(ctx, InspectRequest{Path: "x"}); return err }},
+		{"validate", func(c *ModelRoutingClient, ctx context.Context) error { _, err := c.Validate(ctx, json.RawMessage(`{}`)); return err }},
+		{"apply", func(c *ModelRoutingClient, ctx context.Context) error { _, err := c.Apply(ctx, json.RawMessage(`{}`)); return err }},
+	}
+	for _, tc := range cases {
+		t.Run(tc.op, func(t *testing.T) {
+			var cap []byte
```

golangci-lint (2.12.2)

[error] variable cap has same name as predeclared identifier

(predeclared)